addons/popochiu/editor/gizmos/gizmo_clickable_plugin.gd

```diff
@@ -0,0 +1,241 @@
+				)
+			# TODO: enable this when the look_at_point logic is implemented
+			# LOOK_AT_POINT:
+				# gizmo.set_theme(
+				# 	PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_LOOK_AT_POINT_COLOR),
+				# 	PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_HANDLER_SIZE),
+				# 	default_font,
+				# 	PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_FONT_SIZE)
+				# )
+			BASELINE:
+				gizmo.set_theme(
+					PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_BASELINE_COLOR),
+					PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_HANDLER_SIZE),
+					default_font,
+					PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_FONT_SIZE)
+				)
+
+		gizmo.show_connector = PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_SHOW_CONNECTORS)
+		gizmo.show_outlines = PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_SHOW_OUTLINE)
+		gizmo.show_target_name = PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_SHOW_NODE_NAME)
+		gizmo_id += 1
+	
+	update_overlays()
+#endregion
+
+
+#region Private Methods
+func _update_properties():
+	if _grabbed_gizmo and _grabbed_gizmo.target_property:
+		_target_node.set(
+			_grabbed_gizmo.target_property,
+			_grabbed_gizmo.get_position()
+		)
+
+
+func _init_popochiu_gizmo(gizmo_id: int) -> Gizmo2D:
+	var gizmo: Gizmo2D
+	var default_font = EditorInterface.get_editor_theme().default_font
+
+	match gizmo_id:
+		WALK_TO_POINT:
+			gizmo = Gizmo2D.new(_target_node, "walk_to_point", "Walk To Point", Gizmo2D.GIZMO_POS)
+			gizmo.set_theme(
+				PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_WALK_TO_POINT_COLOR),
+				PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_HANDLER_SIZE),
+				default_font,
+				PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_FONT_SIZE)
+			)
+		# TODO: enable this when the look_at_point logic is implemented
+		# LOOK_AT_POINT:
+		# 	gizmo = Gizmo2D.new(_target_node, "look_at_point", "Look At Point", Gizmo2D.GIZMO_POS)
+		# 	gizmo.set_theme(
+		# 		PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_LOOK_AT_POINT_COLOR),
+		# 		PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_HANDLER_SIZE),
+		# 		default_font,
+		# 		PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_FONT_SIZE)
+		# 	)
+		BASELINE:
+			gizmo = Gizmo2D.new(_target_node, "baseline", "Baseline", Gizmo2D.GIZMO_VPOS)
+			gizmo.set_theme(
+				PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_BASELINE_COLOR),
+				PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_HANDLER_SIZE),
+				default_font,
+				PopochiuEditorConfig._get_editor_setting(PopochiuEditorConfig.GIZMOS_FONT_SIZE)
+			)
+	return gizmo
+
+
+func _try_grab_gizmo(event: InputEvent) -> bool:
```

Comment:
The type of the `event` parameter could be `InputEventMouseButton` since this function will be only called in such input events.